### ISSUE TITLE
Add whenReady API to BrowserWindows to await their set up/rendering

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#23830]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description
I have some code that creates a window (via `window.open`) and then immediately tells the main process to print it. The problem is that the window hasn't finished loading yet / rendering, so it prints a blank page.

I don't really want to have to add a did-finish-load or ready-to-show event for all these windows that I create.

### Proposed Solution
Can we add an API for BrowserWindows (or webcontents?) that returns a promise that resolves once the window has finished loading and renderering? And then we can do:

const win = BrowserWindow.fromId(1);
await win.whenReady();
win.webContents.print(...);

(I guess we'd have to define what exactly `whenReady` means. Maybe use the broad definition of everything has finished loading and rendering)

**Comment by mix3d** [comment]
Actually, if `ready-to-show` DID always fire, resolving immediately when set in cases where the window is ready to show and/or already shown,  you could easily make this with a promise

```JS
function whenReady(window){
  const promise = new Promise()
  window.once('ready-to-show`, ()=>{
    promise.resolve()
  })
return promise
}
```

**Comment by mix3d** [comment]
Though I did just find out that `loadURL()` returns a promise...

**Comment by pushkin-** [comment]
@mix3d Yeah `loadURL` is generally the way to handle this, but in my case, I'm not the one loading the URL into the window. I'm just calling `window.open` and Electron is loading the URL, so I don't have a way to hook into any loadURL promise and have to resort to your method of resolving in the `ready-to-show` event.

**Comment by cawa-93** [comment]
Also will be useful for tests when windows creating in some test-runner wrapper (like playwrite). in my tests I need simply run app and check that window is visible. 

But the electron is actually launched in a child process. And communication between processes and evaluating commands takes time. Therefore, there is a high probability that the event will fire before my check runs. For this reason, I have overhead


```ts
  const window = await electronApp.browserWindow(await electronApp.firstWindow());  // <-- await first created window
  const windowState = await window.evaluate(
    (mainWindow) => {

      return new Promise(resolve => {

        // No way to check did `ready-to-show` event already fired or not!
        mainWindow.once('ready-to-show', () => resolve(mainWindow.isVisible()));

        // If `ready-to-show` event hasn't fired within 10 seconds,
        // I'm just ASSUMING that it actually fired before
        setTimeout(
          () => resolve(mainWindow.isVisible()),
          10_000,
        );

      });
    },
  );
```
